codespaces_guide.md

```diff
@@ -34,13 +41,44 @@ Follow these steps to set up your GitHub Codespace and start coding:
 
    > ⚠️ **For GitHub Pro accounts, including those with the Student Developer Pack, 180 core hours of usage and 20GB of storage are provided for free each month.** For a 2-core setup, this typically allows around 90 hours of usage per month or approximately 3 hours per day. **Please be mindful that any usage beyond this limit _will_ be billed to your account, so use with caution.** However, this amount of time should generally be sufficient to complete the tasks at hand. For more information, please refer to [GitHub Codespaces billing](https://docs.github.com/en/billing/managing-billing-for-github-codespaces/about-billing-for-github-codespaces).
 
-5. Click `Create codespace`. After doing so, a Visual Studio Code interface will appear. The development container environment will be automatically set up, and all necessary Visual Studio Code extensions will be installed. Please allow some time for this process to complete.
+5. Click `Create codespace`. After doing so, a Visual Studio Code interface will appear. The DevContainer environment will be automatically set up, and all necessary Visual Studio Code extensions will be installed. Please allow some time for this process to complete.
 
    > 💡 During the environment setup, you can click on `Building codespace` in the bottom-right corner of the window to view detailed progress logs.
    >
    > 💡 **If an error window appears indicating that your environment is not configured correctly and is running in recovery mode, please delete the Codespace instance you created and try creating a new 2-core instance using the steps above.** If the issue persists, continue this process until you successfully configure a 2-core setup. Usually, this resolves after one or two attempts.
 
- 
+6. **Create and Add an SSH Key:** To create and add an SSH key in the GitHub Codespaces (Visual Studio Code) terminal, follow the instructions starting from Step 5bii in the Windows section of the [README](README.md) for detailed guidance.
+
+7. **Configure Your Repository’s Settings:**
+   1. Update your repository’s remote settings:
+      ```bash
+      git remote rename origin upstream                                              # Rename the existing 'origin' remote to 'upstream'
+      git remote add origin git@github.com:<your_username>/<personal-repo-name>.git  # Add your new private repository as 'origin'
+      git remote set-url --push upstream DISABLE                                     # Disable push access to the public repository
+      ```
+      This configuration allows you to push changes to your private repository while maintaining a reference to the public repository. By renaming the original remote and adding your personal repo as the new `origin`, you ensure that your work is kept private and secure, while still being able to fetch updates from the public repository.
+
+   2. Push to your new repository:
```

Comment:
no need for this step because they already have a private repo with all the commits from the public repo imported